doc/development_process/release_process.rst

```diff
@@ -14,60 +14,58 @@ the features that have actually been implemented, allowing the project to
 maintain the quality of the overall release without delays because of one or two
 features that are not ready yet.
 
-The Zephyr release model is loosely based on the Linux kernel model:
+The Zephyr release model was loosely based on the Linux kernel model:
 
 - Release tagging procedure:
 
   - linear mode on main branch,
   - release branches for maintenance after release tagging.
-- Each release period will consist of a merge window period followed by one or
-  more release candidates on which only stabilization changes, bug fixes, and
+- Each release period will consist of a development phase followed by
+  release candidates on which only stabilization changes, bug fixes, and
   documentation can be merged in.
 
-  - Merge window mode: all changes are accepted (subject to approval from the
+  - Development phase: all changes are accepted (subject to approval from the
     respective maintainers.)
-  - When the merge window is closed, the release owner lays a vN-rc1 tag and the
-    tree enters the release candidate phase
-  - CI sees the tag, builds and runs tests; QA analyses the report from the
-    build and test run and gives an ACK/NAK to the build
-  - The release owner, with QA and any other needed input, determines if the
+  - Stabilisation phase, the release owner lays a vN-rc1 tag and the
+    tree enters the stabilisation phase
+  - CI sees the tag, builds and runs tests; Test teams analyse the report from the
+    build and test run and give an ACK/NAK to the build
+  - The release owner, with test teams and any other needed input, determines if the
     release candidate is a go for release
   - If it is a go for a release, the release owner lays a tag release vN at the
     same point
```

Comment:
It would be worse mentioning about nogo case and vN-rcx iterations.

Reply:
It would also be good to mention bug count status vs quota defined by TSC is one of the main criteria